hack/build-go.sh

```diff
@@ -81,11 +81,13 @@ fi
 
 if [ "${target}" = "install" ]; then
     (
-        mkdir -p ${CMD_OUT_DIR}/container-disk-v2alpha
-        cd cmd/container-disk-v2alpha
-        # the containerdisk bianry needs to be static, as it runs in a scratch container
-        echo "building static binary container-disk"
-        gcc -static -o ${CMD_OUT_DIR}/container-disk-v2alpha/container-disk main.c
+        if [[ $BIN_NAME == *"container-disk"* ]]; then
```

Comment:
BIN_NAME isn't usually set here unless you have some sort of custom build flow going on.

By default, container-disk won't be built if we had this if statement though.

Reply:
I encounter the problem with e.g.

hack/build-go.sh install virt-controller virt-api

Reply:
Hmm, looks like I never mentioned the actual error. Here's a more helpful snippet from the build log

./hack/build-go.sh install cmd/virt-api cmd/virt-controller
 building static binary container-disk
 /usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: cannot find -lc
 collect2: error: ld returned 1 exit status

Reply:
I see. the simpliest work around here is probably 

```if [  -z "$BIN_NAME ] || [[ $BIN_NAME == *"container-disk"* ]]; then```

long term, the structuring of how goland and gcc builds are done needs to be improved.

Reply:
I've added the check for BIN_NAME empty. I also moved this command group to the pre-existing 'if [ "${target}" = "install" ]' conditional. I'll  be happy to change it back if you prefer the group in its own, duplicate conditional.